app/assets/stylesheets/core/_link.scss

```diff
@@ -3,8 +3,7 @@
 @import "../../../node_modules/govuk-frontend/tools/all";
 @import "../../../node_modules/govuk-frontend/core/links";
 
-.app-link--destructive {
-  @include govuk-link-common;
+.govuk-link.app-link--destructive {
```

Comment:
I went through this with @alex-ju already and we think this is the best we can do for now. It's only necessary because our CSS is spread over multiple repos, but I've added a TODO to highlight that it should be fixed :-).